llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -9318,12 +9361,13 @@ void LoopVectorizationPlanner::adjustRecipesForReductions(
            "AnyOf reductions are not allowed for in-loop reductions");
 
     // Collect the chain of "link" recipes for the reduction starting at PhiR.
-    SetVector<VPSingleDefRecipe *> Worklist;
+    SetVector<VPRecipeBase *> Worklist;
     Worklist.insert(PhiR);
     for (unsigned I = 0; I != Worklist.size(); ++I) {
-      VPSingleDefRecipe *Cur = Worklist[I];
-      for (VPUser *U : Cur->users()) {
-        auto *UserRecipe = cast<VPSingleDefRecipe>(U);
```

Comment:
`adjustRecipesForReductions()` may also need to handle multiple defs. Previously, it would `cast` (not `dyn_cast`) that all users were `VPSingleDefRecipe`, so just the presence of a non-`VPSingleDefRecipe` would lead to a crash. 

This patch still only allows it for one defined value (that is not necessarily created by a `VPSingleDefRecipe`), this may need to be changed.